File: 03-07-2025_Find_K-th_Character_in_string_game.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        s = "ab"
        while len(s) < k:
            temp = ''
            for ch in s:
                if ch == 'z':
                    temp += 'a'
                else:
                    temp += chr(ord(ch) + 1)
            s += temp
        return s[k - 1]

File: 03-07-2025_Find_K-th_Character_in_string_game.java
class Solution {
    public char kthCharacter(int k) {
        StringBuilder s = new StringBuilder("ab");
        while (s.length() < k) {
            StringBuilder temp = new StringBuilder();
            for (int i = 0; i < s.length(); i++) {
                char ch = s.charAt(i);
                if (ch == 'z') {
                    temp.append('a');
                } else {
                    temp.append((char)(ch + 1));
                }
            }
            s.append(temp);
        }
        return s.charAt(k - 1);
    }
}
